main.cpp

```diff
@@ -113,19 +141,31 @@ struct <#structname2#>
  Wait for my code review.
  */
 
+
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
-    
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    SoccerBall ballOne( 4.4f, "Wilson");                                             //6
+    SoccerBall ballTwo(5.5f, "Nike");                                             //6
     
-    U <#name3#>;
-    float updatedValue = 5.f;
-    std::cout << "[static func] <#name3#>'s multiplied values: " << <#structname2#>::<#staticFunctionA#>( , ) << std::endl;                  //11
-    
-    U <#name4#>;
-    std::cout << "[member func] <#name4#>'s multiplied values: " << <#name4#>.<#memberFunction#>( &updatedValue ) << std::endl;
+    SoccerBallSize f;                                            //7
+    auto* smaller = f.compare(&ballOne, &ballTwo);                              //8
+    smaller = f.compare(nullptr, nullptr);
+    if(smaller != nullptr)
+    {
+        std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+
+        KickDistance::kickSoccerBall(nullptr, nullptr);
+        
+        StartDistance ballStart1;
+        float updatedValue = 5.0f;
+        std::cout << "kickSoccerBall ballStart's multiplied values: " << KickDistance::kickSoccerBall(&ballStart1, &updatedValue) << std::endl;                  //11
+        
+        StartDistance ballStart2;
+        std::cout << "kickSoccerBall continuousDistance's multiplied values: " << ballStart2.decrementBallDistance(&updatedValue) << std::endl;
+
+        return 0;
+    }
```

Comment:
All of this stuff happens regardless of the value stored in `smaller`
It is not part of the `if(smaller != nullptr)` block
- [ ] revise